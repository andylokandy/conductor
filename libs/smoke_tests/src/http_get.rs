#[cfg(test)]
mod smoke_http_get {
  use insta::assert_debug_snapshot;
  use lazy_static::lazy_static;
  use reqwest::header::{HeaderMap, CONTENT_TYPE};
  use serde_json::Value;
  use std::env::var;

  lazy_static! {
    static ref CONDUCTOR_URL: String = var("CONDUCTOR_URL").expect("CONDUCTOR_URL env var not set");
  }

  #[tokio::test]
  async fn http_get() {
    let mut headers = HeaderMap::default();
    headers.append(
      CONTENT_TYPE,
      "application/x-www-form-urlencoded".parse().unwrap(),
    );

    let req = reqwest::Client::new()
      .request(
        reqwest::Method::GET,
        format!(
          "{}/http-get?query=query%20%7B%20__typename%20%7D",
          CONDUCTOR_URL.as_str()
        ),
      )
      .headers(headers);

    let gql_response = req
      .send()
      .await
      .expect("failed to run http req to conductor");
    assert_eq!(gql_response.status(), 200);
    let json_body = gql_response.json::<Value>().await.unwrap();
    assert_debug_snapshot!(json_body);
  }
}
