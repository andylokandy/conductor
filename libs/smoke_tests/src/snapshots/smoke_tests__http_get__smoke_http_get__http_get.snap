---
source: libs/smoke_tests/src/http_get.rs
expression: json_body
---
Object {
    "data": Object {
        "__typename": String("Query"),
    },
}
